src/gax-internal/src/observability/http_tracing.rs

```diff
@@ -150,7 +148,10 @@ impl HttpSpanInfo {
             otel_name,
             otel_status: OtelStatus::Unset,
             http_request_method: method.to_string(),
-            server_address: url.host_str().map(String::from).unwrap_or_default(),
+            server_address: url
+                .host_str()
+                .map(|h| h.replace("[", "").replace("]", ""))
```

Comment:
Say more words... why?  IPv6?  Just the way `host_str()` is formatted?

Reply:
Consider:

```suggestion
                .map(|h| h.trim_start_matches("["].trim_end_matches("]")).unwrap_or_default().to_string()
```

`.replace()` allocates a new string and has to search through all the bytes.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=ab2418535884562d035c18949ab97058